pkg/client/unversioned/fake/fake.go

```diff
@@ -71,15 +73,23 @@ func (c *RESTClient) Delete() *restclient.Request {
 
 func (c *RESTClient) request(verb string) *restclient.Request {
 	config := restclient.ContentConfig{
-		ContentType:  runtime.ContentTypeJSON,
-		GroupVersion: testapi.Default.GroupVersion(),
-		Codec:        c.Codec,
+		ContentType:          runtime.ContentTypeJSON,
+		GroupVersion:         testapi.Default.GroupVersion(),
+		NegotiatedSerializer: c.NegotiatedSerializer,
+	}
+	ns := c.NegotiatedSerializer
```

Comment:
I'm kind of worried about doing this per request - rather than per client before.  It makes sense, but it's also really wasteful.  What other options did you consider?